src/parachain/loans.ts

```diff
@@ -335,6 +369,77 @@ export class DefaultLoansAPI implements LoansAPI {
         return this._getPositionsOfAccount(accountId, this._getBorrowPosition.bind(this));
     }
 
+    getLoanCollateralInfo(
+        lendPositions: Array<LendPosition>,
+        borrowPositions: Array<BorrowPosition>,
+        loanAssets: TickerToData<LoanAsset>
+    ): LoanCollateralInfo {
+        const lendCollateralPositions = lendPositions.filter(({ isCollateral }) => isCollateral);
+        const lendCollateralThresholdAdjustedPositions = lendCollateralPositions.map((position) => {
+            const collateralTheshold = loanAssets[position.currency.ticker].collateralThreshold;
```

Comment:
> loanAssets[position.currency.ticker]

There's a chance this will throw an error if the currency ticker changes (more likely for foreign assets). It's unlikely, but you should add a doc comment about it at least, so the UI wraps this in a try/catch.

Reply:
This can also occur if the `loanAssets` parameter is empty (for whatever reason) while `lendCollateralPositions` is non-empty

Reply:
Actually, this function has to throw if `loanAssets` object does not contain data for some of the currencies that loan positions consist of. I added function to check this and return meaningful error message if such case occurs.